presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/VerifyNoApplyNodeLeftPlanOptimizer.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.optimizations;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.StandardErrorCode;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.planner.PlanNodeIdAllocator;
+import com.facebook.presto.sql.planner.SimplePlanVisitor;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.SymbolAllocator;
+import com.facebook.presto.sql.planner.plan.ApplyNode;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+
+import java.util.Map;
+
+public class VerifyNoApplyNodeLeftPlanOptimizer
```

Comment:
It used to be written as sanity checker. 

Plan sanity checkers are run before optimizers where apply nodes still exist. I could change sanity checkers to accept parameter flag which would tell if plan is optimized or not, but to make this work all other optimizers would need to support Apply node. Otherwise, instead of getting an error from sanity checkers we could get some mysterious error from other optimizer rule.

I don't like it either but I don't have a better idea for this.